### ISSUE TITLE
rustc --version reported date is confusing

### ISSUE THREAD

**Issue by brson** [rust-lang/rust#21957]
I've heard several people express concern that their nightly was not up to date because the date reported by `rustc --version` is old. This date is the commit date, not the build date.

**Comment by brson** [comment]
cc @eddyb

**Comment by brson** [comment]
On this subject @eddyb said

```
22:31 < eddyb_> brson: would be nice if it gave build info and source (git) info
22:31 < eddyb_> but clearly distinguished from each-other
```

to which blaenk responded

```
22:35 < blaenk> yeah that'd be cool
```

**Comment by brson** [comment]
Easy to implement. Just need to decide what to change.

**Comment by huonw** [comment]
FWIW, even the (start of) build date may be confusing, since the nightly builds take many hours.

**Comment by brson** [comment]
Should update how rustdoc and cargo report their version at the same time.

**Comment by iKevinY** [comment]
Looks like this can be closed since #22201 has been rolled up.
